tracer/src/Datadog.Trace/SpanContextExtractor.cs

```diff
@@ -0,0 +1,31 @@
+// <copyright file="SpanContextExtractor.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Propagators;
+
+#nullable enable
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// The SpanContextGetter is responsible for extracting SpanContext in the rare cases where the Tracer couldn't propagate it itself.
+    /// This can happen for instance when libraries add an extra layer above the instrumented ones
+    /// (eg consuming Kafka messages and enqueuing them prior to generate a span).
+    /// </summary>
```

Comment:
I'll re-recommend Andrew's suggestion:

```suggestion
    /// <inheritdoc />
```